### PR TITLE
nodejs: corepack wrappers package

### PR DESCRIPTION
**Pull request by wmertens**
###### Description of changes

Corepack is the nodejs effort to put version on package managers in package.json. This package adds symlinks to the package managers that will use the version specified in `packageManager`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wmertens** [comment]
Maybe @SuperSandro2000 can take a quick look to say if it is the correct approach?

File: pkgs/development/web/nodejs/corepack.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    corepack enable --install-directory $out/bin
+    # Also wrap npm
+    corepack enable --install-directory $out/bin npm
+  '';
+
+  meta = {
+    description = "Wrappers for npm, pnpm and yarn via nodejs's corepack";
```
Comment (aduh95) [inline pkgs/development/web/nodejs/corepack.nix:21]: nit

```suggestion
    description = "Wrappers for npm, pnpm and Yarn via Node.js Corepack";
```

refs: https://github.com/nodejs/node/blob/cc80513551503397e07c46f84e57ea48c342561e/doc/README.md#L87-L89

File: pkgs/development/web/nodejs/corepack.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, nodejs }:
+
+let
+  inherit (nodejs) version;
+in
+stdenv.mkDerivation {
+  name = "corepack-nodejs-${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/web/nodejs/corepack.nix:7]:
```suggestion
  pname = "corepack-nodejs";
  inherit (nodejs) version;
```

File: pkgs/development/web/nodejs/corepack.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, nodejs }:
+
+let
+  inherit (nodejs) version;
+in
```
Comment (SuperSandro2000) [inline pkgs/development/web/nodejs/corepack.nix:5]:
```suggestion
```

File: pkgs/development/web/nodejs/corepack.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+stdenv.mkDerivation {
+  name = "corepack-nodejs-${version}";
+
+  nativeBuildInputs = [ nodejs ];
+
+  unpackPhase = "true";
```
Comment (SuperSandro2000) [inline pkgs/development/web/nodejs/corepack.nix:11]:
```suggestion
  dontUnpack = true;
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -9864,6 +9864,8 @@ with pkgs;
   nodenv = callPackage ../development/tools/nodenv { };
 
   nodejs = hiPrio nodejs_18;
+  corepack = hiPrio (callPackage ../development/web/nodejs/corepack.nix {
+    nodejs = nodejs; });
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:9868]:
```suggestion
  corepack = hiPrio (callPackage ../development/web/nodejs/corepack.nix { });
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -9882,13 +9884,20 @@ with pkgs;
...
     enableNpm = false;
   };
+  corepack_20 = hiPrio (callPackage ../development/web/nodejs/corepack.nix
+  { nodejs = nodejs_20; });
   # Update this when adding the newest nodejs major version!
   nodejs_latest = nodejs_20;
   nodejs-slim_latest = nodejs-slim_20;
+  corepack_latest = hiPrio (callPackage ../development/web/nodejs/corepack.nix
+  { nodejs = nodejs_latest; });
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:9900]:
```suggestion
  corepack_18 = hiPrio (callPackage ../development/web/nodejs/corepack.nix { nodejs = nodejs_18; });

  nodejs_20 = callPackage ../development/web/nodejs/v20.nix { };
  nodejs-slim_20 = callPackage ../development/web/nodejs/v20.nix {
    enableNpm = false;
  };
  corepack_20 = hiPrio (callPackage ../development/web/nodejs/corepack.nix { nodejs = nodejs_20; });

  # Update this when adding the newest nodejs major version!
  nodejs_latest = nodejs_20;
  nodejs-slim_latest = nodejs-slim_20;
  corepack_latest = hiPrio (callPackage ../development/web/nodejs/corepack.nix { nodejs = nodejs_latest; });
```

File: pkgs/development/web/nodejs/corepack.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    corepack enable --install-directory $out/bin
+    # Also wrap npm
+    corepack enable --install-directory $out/bin npm
+  '';
+
+  meta = {
+    description = "Wrappers for npm, pnpm and yarn via nodejs's corepack";
+    homepage = "https://nodejs.org";
```
Comment (SuperSandro2000) [inline pkgs/development/web/nodejs/corepack.nix:22]:
```suggestion
    homepage = "https://nodejs.org/api/corepack.html";
```

**Pull request merged**